go-controller/pkg/ovn/master.go

```diff
@@ -121,6 +124,81 @@ func (oc *Controller) Start(kClient kubernetes.Interface, nodeName string, wg *s
 	return nil
 }
 
+// delete obsoleted logical OVN entities that are specific for Multiple join switches OVN topology. Also cleanup
+// OVN entities for deleted nodes (similiar to syncNodes() but for obsoleted Multiple join switches OVN topology
+func (oc *Controller) upgradeToSingleSwitchOVNTopology(existingNodeList *kapi.NodeList) error {
+
+	existingNodes := make(map[string]bool)
+	for _, node := range existingNodeList.Items {
+		existingNodes[node.Name] = true
+
+		// delete the obsoleted node-join-subnets annotation
+		err := oc.kube.SetAnnotationsOnNode(&node, map[string]interface{}{"k8s.ovn.org/node-join-subnets": nil})
+		if err != nil {
+			klog.Errorf("Failed to remove node-join-subnets annotation for node %s", node.Name)
+		}
+	}
+
+	nodeSwitches, stderr, err := util.RunOVNNbctl("--data=bare", "--no-heading",
+		"--columns=name,other-config", "find", "logical_switch", "other-config:subnet!=_")
```

Comment:
what is the significance of matching on "other-config:subnet!=_" ?

Reply:
Because there are logical switches (join, ext-<nodeName etc.) that do not have other-config options, the output of the command would not be as expected and parsing will be incorrect. I was trying to filter out those swithes. But my changes are not correct either, as the node switch may not have "other-config:ipv6_prefix" set, not "other-config:subnet".